src/audio/mfcc/mfcc_generic.c

```diff
@@ -0,0 +1,444 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <sof/audio/mfcc/mfcc_comp.h>
+#include <sof/audio/component.h>
+#include <sof/audio/audio_stream.h>
+#include <sof/math/auditory.h>
+#include <sof/math/matrix.h>
+#include <sof/math/sqrt.h>
+#include <sof/math/trig.h>
+#include <sof/math/window.h>
+#include <sof/trace/trace.h>
+#include <user/mfcc.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/* MFCC with 16 bit FFT benefits from data normalize, for 32 bits there's no
+ * significant impact. The amount of left shifts for FFT input is limited to
+ * 10 that equals about 60 dB boost. The boost is compensated in Mel energy
+ * calculation.
+ */
+#if MFCC_FFT_BITS == 16
+#define MFCC_NORMALIZE_FFT
+#else
+#undef MFCC_NORMALIZE_FFT
+#endif
+#define MFCC_NORMALIZE_MAX_SHIFT	10
+
+/* Open files for data debug output in testbench */
+#define DEBUG_WITH_TESTBENCH
+#if defined(CONFIG_LIBRARY) && defined(DEBUG_WITH_TESTBENCH)
+#define DEBUGFILES
+#endif
+
+#ifdef MFCC_DEBUGFILES
```

Comment:
can this be done as a cmocka test?

Reply:
Yes, I will move libraries tests to cmocka after they have been split out of this main PR. The MFCC overall test will be testbench based. Like with TDFB I will attempt to extract needed internal state information from traces instead of these files. So, these will go away.

Reply:
I was thinking to add cmocka unit test with other PR since this is quite large. It could be part of #5769 that creates the reference data with Pytorch to compare. Would that be OK @lgirdwood ?